From b7789bb144ff4545021fbd95ee93c98ec706891a Mon Sep 17 00:00:00 2001
From: Nicola Murino <nicola.murino@gmail.com>
Date: Fri, 17 Aug 2018 17:04:07 +0200
Subject: [PATCH] gsocket: fix cross compilation

for some reason when cross compiling for windows BROKEN_IP_MREQ_SOURCE_STRUCT is defined but should not
---
 gio/gsocket.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gio/gsocket.c b/gio/gsocket.c
index 859e807cb..71a97b8cf 100644
--- a/gio/gsocket.c
+++ b/gio/gsocket.c
@@ -2404,7 +2404,7 @@ g_socket_multicast_group_operation_ssm (GSocket       *socket,
         memset (&mc_req_src, 0, sizeof (mc_req_src));
 
         /* By default use the default IPv4 multicast interface. */
-        S_ADDR_FIELD(mc_req_src) = g_htonl (INADDR_ANY);
+	mc_req_src.imr_interface.s_addr = g_htonl (INADDR_ANY);
 
         if (iface)
           {
-- 
2.18.0

